### PR TITLE
AWS sns topic support sub attributes

### LINKED ISSUE #51756: SNS Topic Subscription filter_policy
##### SUMMARY
SNS Topic (sns_topic) in Ansible supports subscriptions but it doesn't seem to support subscription filter_policies. 

https://docs.aws.amazon.com/sns/latest/dg/sns-message-filtering.html 

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
sns_topic subscriptions. 

##### ADDITIONAL INFORMATION

This feature is important so messages sent to the SNS topic subscriber (Lambda, SQS, etc..) are filtered at the AWS level before it reaches the code or queue subscribed to the topic. 

```yaml

  subscriptions:
    - endpoint: "{{ aws_lambda_result.configuration.function_arn}}"
       protocol: lambda
       filter_policy:
         attribute_a:
           - value_1
           - value_2
         attribute_b:
           - value_3



```

### PR DESCRIPTION
**Pull request by brianhhq**
##### SUMMARY
Adding the ability to set subscription attributes when subscribing to sns topic.
Use object attribute to track the changed status

Fixes #51756

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/amazon/sns_topic.py

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by wimnat** [comment]
LGTM

shipit

**Comment by brianhhq** [comment]
hi @joelthompson @nand0p @willthames would you please review and merge this PR?

**Comment by brianhhq** [comment]
hi @joelthompson @nand0p @willthames would you please review and merge this PR?

**Comment by brianhhq** [comment]
hi @joelthompson @nand0p @willthames would you please review and merge this PR?

**Comment by brianhhq** [comment]
bot_status

**Comment by brianhhq** [comment]
ready_for_review

File: lib/ansible/modules/cloud/amazon/sns_topic.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
   - "Joel Thompson (@joelthompson)"
   - "Fernando Jose Pando (@nand0p)"
   - "Will Thames (@willthames)"
+  - "Brian Huang (@brianhhq)"
```
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/sns_topic.py:26]:
```suggestion
```
 Thanks very much for this new feature. In the Ansible project, module authors are our community maintainers who are granted the ability to issue [shipits](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md#community) for changes, basically giving these individuals merge powers. As this is not a new module or a substantial rewrite, this line isn't needed here. We'd like to have you get more involved with the project before granting approval capabilities on modules.
Are there certain modules you are particularly interested in?
Reviewing PRs really helps us see [AWS priorities](https://github.com/ansible/community/wiki/AWS%3A-Pull-Request-Priorities) for a list of GitHub searches.

**Review by jillr — COMMENTED** [review]
Could you please add [integration tests](https://docs.ansible.com/ansible/devel/dev_guide/platforms/aws_guidelines.html#integration-tests-for-aws-modules) for this new feature?  They would be added to the existing test suite in `test/integration/targets/sns_topic/`.  We generally look for new tests to cover the functionality provided by the change with:
Resource creation under check mode
Resource creation
Resource creation again (idempotency)
Resource modification
Resource deletion

**Pull request closed without merging**